### PR TITLE
Add module mmseqs easysearch

### PR DESCRIPTION
**Pull request by luisas**
Add module mmseqs easysearch

Gets as input a fasta file and a database. 
Outputs a tsv file with the summary of the matches. 

## PR checklist

Closes https://github.com/nf-core/modules/issues/3698

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/mmseqs/easysearch/main.nf [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        'https://depot.galaxyproject.org/singularity/mmseqs2:14.7e284--pl5321h6a68c12_2':
+        'biocontainers/mmseqs2:14.7e284--pl5321h6a68c12_2' }"
+
+    input:
+    tuple val(meta),  path(fasta)
+    tuple val(meta2), path(db_target)
+
+    output:
+    tuple val(meta), path("${prefix}.tsv"), emit: tsv
+    path "versions.yml"               , emit: versions
```
Comment (Joon-Klaps) [inline modules/nf-core/mmseqs/easysearch/main.nf:17]:
```suggestion
    path "versions.yml"                   , emit: versions
```

File: tests/modules/nf-core/mmseqs/easysearch/nextflow.config [hunk 2]
```diff
@@ -0,0 +1,8 @@
+process {
+
+    publishDir = { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" }
+
```
Comment (Joon-Klaps) [inline tests/modules/nf-core/mmseqs/easysearch/nextflow.config:4]: A small suggestion to clean up the output test.yml 
```suggestion
    withName: MMSEQS_CREATEDB_TARGET {
        publishDir = [ enabled: false ]
    }

```

File: modules/nf-core/mmseqs/easysearch/main.nf (hunk 1 above)
Comment (Joon-Klaps) [inline modules/nf-core/mmseqs/easysearch/main.nf:17]:
```suggestion
    path "versions.yml"                   , emit: versions
```

**Comment by Joon-Klaps** [comment]
LGTM, just a small edit regarding [the alignment](https://nf-co.re/docs/contributing/modules#what-is-the-harshil-alignment) and a small suggestion to clean up the test yml 🚀

**Review by Joon-Klaps — CHANGES_REQUESTED** [review]
LGTM, just a small edit regarding [the alignment](https://nf-co.re/docs/contributing/modules#what-is-the-harshil-alignment) and a small suggestion to clean up the test yml. Rerun test and 🚀

File: modules/nf-core/mmseqs/easysearch/main.nf [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    tag "$meta.id"
+    label 'process_high'
+
+    conda "bioconda::mmseqs2=14.7e284"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mmseqs2:14.7e284--pl5321h6a68c12_2':
+        'biocontainers/mmseqs2:14.7e284--pl5321h6a68c12_2' }"
+
+    input:
+    tuple val(meta),  path(fasta)
```
Comment (Joon-Klaps) [inline modules/nf-core/mmseqs/easysearch/main.nf:12]:
```suggestion
    tuple val(meta) , path(fasta)
```

File: modules/nf-core/mmseqs/easysearch/main.nf [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    mkdir -p ${prefix}
+
+    # Extract files with specified args based suffix | remove suffix | isolate longest common substring of files
+    DB_TARGET_PATH_NAME=\$(find -L "$db_target/" -maxdepth 1 -name "*.dbtype" | sed 's/\\.\\[^.\\]*\$//' | sed -e 'N;s/^\\(.*\\).*\\n\\1.*\$/\\1\\n\\1/;D' )
```
Comment (Joon-Klaps) [inline modules/nf-core/mmseqs/easysearch/main.nf:29]:
```suggestion
    DB_TARGET_PATH_NAME=\$(find -L "$db_target/" -maxdepth 1 -name "$args2" | sed 's/\\.\\[^.\\]*\$//' | sed -e 'N;s/^\\(.*\\).*\\n\\1.*\$/\\1\\n\\1/;D' )
```

**Review by Joon-Klaps — CHANGES_REQUESTED** [review]
Don't forget to rerun the `nf-core modules create-test-yml -t mmseqs/easysearch`

**Comment by luisas** [comment]
> Don't forget to rerun the `nf-core modules create-test-yml -t mmseqs/easysearch`

Thanks, done now :)

File: modules/nf-core/mmseqs/easysearch/main.nf [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        mmseqs: \$(mmseqs | grep 'Version' | sed 's/MMseqs2 Version: //')
+    END_VERSIONS
+    """
+
+    stub:
+    def args = task.ext.args ?: ''
```
Comment (Joon-Klaps) [inline modules/nf-core/mmseqs/easysearch/main.nf:50]:
```suggestion
    def args = task.ext.args ?: ''
    def args2 = task.ext.args2 ?: "*.dbtype"
```

**Review by Joon-Klaps — APPROVED** [review]
Great work, thanks! 👍

**Pull request merged**